File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/resources/StartRegistreringStatusMetrikker.java
package no.nav.fo.veilarbregistrering.registrering.bruker.resources;

import no.nav.fo.veilarbregistrering.metrics.Event;
import no.nav.fo.veilarbregistrering.metrics.Metric;
import no.nav.fo.veilarbregistrering.metrics.MetricsService;
import no.nav.fo.veilarbregistrering.registrering.bruker.RegistreringType;


class StartRegistreringStatusMetrikker {
    static void rapporterRegistreringsstatus(MetricsService metricsService, StartRegistreringStatusDto registreringStatus) {
        metricsService.reportFields(
                Event.of("registrering.bruker.data"),
                Metric.of("erAktivIArena", registreringStatus.getRegistreringType() == RegistreringType.ALLEREDE_REGISTRERT),
                Metric.of("kreverReaktivering", registreringStatus.getRegistreringType() == RegistreringType.REAKTIVERING),
                Metric.of("sykmeldUnder39uker", registreringStatus.getRegistreringType() == RegistreringType.SPERRET),
                Metric.of("sykmeldOver39uker", registreringStatus.getRegistreringType() == RegistreringType.SYKMELDT_REGISTRERING),
                Metric.of("jobbetSiste6av12Mnd", registreringStatus.getJobbetSeksAvTolvSisteManeder())
        );

    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/resources/StartRegistreringStatusMetrikker.kt
package no.nav.fo.veilarbregistrering.registrering.bruker.resources

import no.nav.fo.veilarbregistrering.metrics.Event.Companion.of
import no.nav.fo.veilarbregistrering.metrics.Metric.Companion.of
import no.nav.fo.veilarbregistrering.metrics.InfluxMetricsService
import no.nav.fo.veilarbregistrering.registrering.bruker.RegistreringType

internal object StartRegistreringStatusMetrikker {
    @JvmStatic
    fun rapporterRegistreringsstatus(influxMetricsService: InfluxMetricsService, registreringStatus: StartRegistreringStatusDto) {
        var fields = listOf(
            of("erAktivIArena", registreringStatus.registreringType == RegistreringType.ALLEREDE_REGISTRERT),
            of("kreverReaktivering", registreringStatus.registreringType == RegistreringType.REAKTIVERING),
            of("sykmeldOver39uker", registreringStatus.registreringType == RegistreringType.SYKMELDT_REGISTRERING)
        )

        registreringStatus.jobbetSeksAvTolvSisteManeder?.let {
            fields = fields + of("jobbetSiste6av12Mnd", it)
        }

        influxMetricsService.reportFields(
            of("registrering.bruker.data"),
            *fields.toTypedArray()
        )
    }
}